### ISSUE TITLE
Consideration for Contributors with Limited Internet Access: Partial Clone and Shallow Clone Support

### ISSUE THREAD

**Issue by richarddushime** [servo/servo#31637]
The size of the Servo repository poses a potential barrier to contribution for individuals with limited internet connectivity. Even those with moderately good internet connections may find it cumbersome to clone the entire repository for making minor edits or contributions.

While modifying the git history to reduce repository size might be a viable approach for some projects, it's not feasible for Servo due to its complexity and extensive history.

**Proposed Solution:**

To address this issue and make contributing to Servo more accessible, we should consider providing support and guidance for partial clone and shallow clone techniques. By incorporating details on these techniques in the contributing guidelines or documentation, we can make contributors with limited internet access to make contributions more efficiently.

**Additional Details:**

Partial clones and shallow clones allow users to clone only a portion of a repository's history, significantly reducing the amount of data downloaded during the cloning process. This can be particularly helpful for contributors who only need to make changes to specific parts of the codebase too.

**References:**

[GitHub Blog - Get up to speed with partial clone and shallow clone](https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/)

**Comment by richarddushime** [comment]
Hi @mrobinson @Loirooriol 
I came to this realization while setting up the project, which ended up taking me a significant amount of time.
i would be happy to work on it if accepted

**Comment by Loirooriol** [comment]
Adding something to the README seems good to me. Having less objects also means faster git operations and less disk space. At the moment I'm using a `.git/shallow` with

```
06873da56625640f252814707befa96bf130dad4
095f22c314130e019d3b8eb7e3f268a0cfce51b8
ce57256327fbb0ac20cefd796440b421951012ad
d577d4489a22b0b7cf3c12bf6cabe2ecb5492bcc
```

**Comment by richarddushime** [comment]
i have playing around  with all  the shallow options 
both  `git clone --depth=1`  and `--filter=blob:none `   works and are good  but  `--filter=blob:none `  is said to be Good for developers

**Comment by richarddushime** [comment]
@mrobinson should i go ahead and work on this ?

**Comment by mrobinson** [comment]
Sure. I think this information could go on https://github.com/servo/servo/wiki/Building#troubleshooting-the-windows-build. You can write it here and then we can integrate it. Then once it's posted we can include a link to it from the README.

**Comment by Loirooriol** [comment]
But this isn't windows-specific

**Comment by mrobinson** [comment]
> But this isn't windows-specific

Sorry. I posted the wrong link. I meant https://github.com/servo/servo/wiki/Building.

**Comment by richarddushime** [comment]
> > But this isn't windows-specific
> 
> Sorry. I posted the wrong link. I meant https://github.com/servo/servo/wiki/Building.

I f understand what you meant , I should create a separate section in https://github.com/servo/servo/wiki/Building 

then  I should add a link in the main Readme  that leads to the section 

Alternative 
I was thinking I should add it  something like  below in the Readme.md File on the section for cloning :
[https://github.com/servo/servo/pull/31697/files](https://github.com/servo/servo/pull/31697/files)

**Comment by sagudev** [comment]
SIDE-NOTE: There is also checkout step for some git dependencies in cargo, but it's currently not possible to set it to do shallow clone (`-Zgit` will become available in rust 1.77: https://github.com/rust-lang/cargo/pull/13252).
